R/ML.SpeedGLMSGD.R

```diff
@@ -0,0 +1,240 @@
+#' ML.SpeedGLMSGD
+#'
+#' Base class for Speedglm in combination with SGD machine learning model.
+#'
+#' @docType class
+#' @importFrom R6 R6Class
+#' @importFrom Speedglm speedglm
+#' @importFrom MASS write.matrix
+#' @include ML.Base.R
+#' @section Methods:
+#' \describe{
+#'   \item{\code{initialize(weights = NULL, 
+#'offset = NULL, 
+#'sparse = NULL, 
+#'set.default = list(), 
+#'method = c("eigen", "Cholesky", "qr"), 
+#'model = FALSE, 
+#' y = FALSE, 
+#'fitted = FALSE, 
+#'subset = NULL
+#'family=binomial(link=logit)
+#'
+
+#'
+#'     @param family string the statistic algorithm used to
+#'      optimize.
+#'
+#'   }
+#'
+#'   \item{\code{get_nthread}}{
+#'     Active method. Function that returns the number of threads the speedlm
+#'     algorithm runs on.
+#'   }
+#'
+#'   \item{\code{get_validity}}{
+#'     Active method. Function that shows wheter the current configuration of
+#'     the initialisation is valid. The function returns \code{TRUE} if everything is
+#'     specified correctly. It will throw an error (with the error messages)
+#'     when something is misspecified. This function is automatically called
+#'     after initialization.
+#'   }
+#' }
+#' 
+#' ## TO DO implement 
+ML.SpeedGLMSGD <- R6Class("ML.SpeedGLMSGD",
+  inherit = ML.Base,
+  public =
+    list(
+      fitfunname='speedglm',
+      lmclass='speedglm',
+      initialize = function() {
+        self$get_validity
+        super$initialize()
+        
+      }
+    ),  active =
+    list(get_file_name = function() {
+         return(private$file_name)
+    }
+    
+      ),
+  private =
+      list(
+           alpha = 0.001,
+           loop = 1000,
+           file_name = file.path('output', 'coef.prn'),
+      
+      do.predict = function(X_mat,Y_vals,save_model=FALSE,m.fit=NULL) {
+        X_mat <- data.frame(X_mat)
+        Y_vals <-data.frame(Y_vals)
+        data<-cbind.data.frame(Y_vals,X_mat)
+        
+        #rename colums to standardize depending on the number of columns
+        for (i in (1:(ncol(data)))){
```

Comment:
this loop can be extracted into a function, so you don't have to repeat it. Also, I'm not sure why we would need to do this. I remember you told me that it has to do with being consistent, but in fact you could argue that you're not consistent, as we don't perform this transformation in the other classes.